microservices/resource-management-control/src/service/service.js

```diff
@@ -1,54 +1,101 @@
+/* eslint-disable-next-line no-unused-vars */
+const jwt = require('jsonwebtoken');
 
 exports.createLecture = async (call, callback) => {
-    const { courseId, name, url } = call.request;
-
-    const resource = {
-        path: url,
-    };
-    await getRepository(Resource).save(resource);
-    const madeResource = await getRepository(Resource)
-        .createQueryBuilder('resource')
-        .where('resource.path = :path', { path: call.request.url })
-        .getOne();
-    const resourceId = madeResource.id;
-
-    const lecture = {
-        name,
-        resourceId,
-        courseId,
-    };
-    console.log(lecture);
-
-    await getConnection()
-        .createQueryBuilder()
-        .insert()
-        .into(Lecture)
-        .values([{
-            name: lecture.name,
-            resourceId: lecture.resourceId,
-            courseId: lecture.courseId,
-        }])
-        .execute();
-
-    callback(null);
+    const { courseId, name } = call.request;
+
+    let resource;
+    try {
+        resource = await getConnection()
+            .createQueryBuilder()
+            .insert()
+            .into(Resource)
+            .values([])
+            .execute();
+    } catch (error) {
+        const status = grpc.status.INTERNAL;
+        return errorHandler(callback, status, 'Database error', error);
+    }
+    const resourceId = resource.raw[0].id;
+
+    const token = jwt.sign(
+        { resourceId: resourceId },
+        process.env.JWT_SECRET,
+        {
+            expiresIn: process.env.JWT_VALID_TIME
+        }
+    );
+    const url = process.env.RESOURCE_URL + '/upload/' + token;
+
+    try {
+        await getConnection()
+            .createQueryBuilder()
+            .insert()
+            .into(Lecture)
+            .values([
+                { name: name, resource: resourceId, course: courseId },
+            ])
+            .execute();
+    } catch (error) {
+        const status = grpc.status.INTERNAL;
+        return errorHandler(callback, status, 'Database error', error);
+    }
+
+    callback(null, {url});
 };
 
-exports.getAllLectures = async (call, callback) => {
-    const lectures = await getRepository(Lecture)
-        .createQueryBuilder('lecture')
-        .where('lecture.course_id = :courseId',
-            { courseId: call.request.courseId })
-        .getMany();
+exports.getLecture = async (call, callback) => {
+    const { id } = call.request;
 
-    console.log(lectures);
+    const lectureRepository = getRepository(Lecture);
 
+    let lecture;
+    try {
+        lecture = await lectureRepository.findOne(id);
+    } catch (error) {
+        const status = grpc.status.INTERNAL;
+        return errorHandler(callback, status, 'Database error', error);
+    }
+    if (lecture == null) {
+        const status = grpc.status.INVALID_ARGUMENT;
+        return errorHandler(callback, status, 'No lecture with this id');
+    }
+
+    const token = jwt.sign(
+        { resourceId: lecture.resourceId },
```

Comment:
I hope you have fixed column name to be resource_id

Reply:
i Did